lib/sidebar/index.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import type { IView } from '../navigation/view.ts'
+import type { IFolder, INode } from '../node/index.ts'
+
+import isSvg from 'is-svg'
+import logger from '../utils/logger.ts'
+
+export interface ISidebarContext {
+	/**
+	 * The active node in the sidebar
+	 */
+	node: INode
+
+	/**
+	 * The current open folder in the files app
+	 */
+	folder: IFolder
+
+	/**
+	 * The currently active view
+	 */
+	view: IView
+}
+
+/**
+ * This component describes the custom web component that should be registered for a sidebar tab.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Web_components
+ * @see https://vuejs.org/guide/extras/web-components#building-custom-elements-with-vue
+ */
+export type SidebarComponent = HTMLElement & ISidebarContext
+
+/**
+ * Implementation of a custom sidebar tab within the files app.
+ */
+export interface ISidebarTab {
+	/**
+	 * Unique id of the sidebar tab.
+	 * This has to conform to the HTML id attribute specification.
+	 */
+	id: string
+
+	/**
+	 * The localized name of the sidebar tab.
+	 */
+	displayName: string
+
+	/**
+	 * The icon, as SVG, of the sidebar tab.
+	 */
+	iconSvg: string
+
+	/**
+	 * The order of this tab.
+	 * Use a low number to make this tab ordered in front.
+	 */
+	order: number
+
+	/**
+	 * Name of the web component as used to register it.
+	 * The web component must already be defined with `CustomElementRegistry.define()`.
+	 *
+	 * To avoid name clashes the name has to start with your appid (e.g. `your_app`).
+	 * So in addition with the web component naming rules a good name would be `your_app-files-sidebar-tab`.
+	 */
+	component: string
```

Comment:
On viewer I went for something like `tagname` which I think is more precise than component. Thoughts ?